operator/v1/types_console.go

```diff
@@ -143,8 +143,107 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeMode is the value of the Logo Theme Mode field that determines the theme mode in the console UI.
+type ThemeMode string
+
+// ThemeMode values
+const (
+	// ThemeModeDark represents the dark theme for the console UI.
+	ThemeModeDark ThemeMode = "Dark"
+
+	// ThemeModeLight represents the light theme for the console UI.
+	ThemeModeLight ThemeMode = "Light"
+)
+
+// LogoType is the value of the Logo Type field that determines if the logo is for the masthead or the favicon in the console UI.
+// The masthead logo is displayed in the masthead and about modal of the console UI.
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// LogoFileReference references a specific file within a ConfigMap.
+type LogoFileReference struct {
+	// name is the name of the ConfigMap.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9\-_\.]+$`
```

Comment:
We generally recommend using CEL validations instead of patterns now so that we can return a helpful message to the user if this fails validation.

Reply:
Also, make sure you include any validations in the GoDoc.

Reply:
> We generally recommend using CEL validations instead of patterns now so that we can return a helpful message to the user if this fails validation.

Could you share an example of using CEL validations instead of patterns in the code repository?

Reply:
You have one right here: https://github.com/openshift/api/pull/2177/files#diff-2d91c76ad45d9598ff5cf4b5a091f9641507a937859fca78eadcece3e6cae9e9R181 :)

Reply:
> You have one right here: https://github.com/openshift/api/pull/2177/files#diff-2d91c76ad45d9598ff5cf4b5a091f9641507a937859fca78eadcece3e6cae9e9R181 :)

I don't understand how this applies here.". This is a validation for Enum `// +kubebuilder:validation:Enum="Dark";"Light"`, right?

Reply:
This might work // +kubebuilder:validation:XValidation:rule="self.matches('^[a-zA-Z0-9\\-_\\.]+$')",message="Name must consist of alphanumeric characters, dashes, underscores, or dots." Name string json:"name"``. 
But I didn't see any similar validation in the code for strings. The `metadata.name` is a commonly used property, so there should be a similar validation for it if this is something that is recommended instead of rewriting it."
cc: @jhadvig

Reply:
@cajieh yes we should be using the `XValidation` foe these fields.

`// +kubebuilder:validation:XValidation:rule="self.matches('^[a-zA-Z0-9\\-_\\.]+$')",message="The ConfigMap name must contain only alphanumeric characters, dashes (-), underscores (_), or dots (.)"`

and

`// +kubebuilder:validation:XValidation:rule="self.matches('.*\\.(png|jpg|jpeg|svg|gif)$')",message="The ConfigMap key must end with a valid image file extension (.png, .jpg, .jpeg, .svg, .gif)."`